redwood-protocol-host/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt

```diff
@@ -56,13 +57,35 @@ public class ProtocolBridge<W : Any>(
   )
   private val changedWidgets = mutableSetOf<ChangeListener>()
 
+  /** Nodes available for reuse. */
+  private val pool = ArrayDeque<ProtocolNode<W>>()
+
   override fun sendChanges(changes: List<Change>) {
+    @Suppress("NAME_SHADOWING")
+    val changes = applyReuse(changes)
+      .sortedBy {
+        // Create nodes, set their properties, set their modifiers, and then finally put them in
+        // their place in the widget tree. This order is chosen such that:
+        //  - no redundant modifier updates on newly-created children
+        //  - we don't apply modifiers until their properties are all set
+        //  - we can't change the relative orders of insert/move/remove without adjusting indexes
+        when (it) {
+          is Create -> 0
+          is PropertyChange -> 1
+          is ModifierChange -> 2
+          else -> 3
+        }
+      }
```

Comment:
This order is already guaranteed by the protocol with the exception of modifiers coming before properties. If you want to invert that order, we can change the generated Compose code to put the modifier update last in the list.

This will retain symmetric behavior between Redwood and protocol Redwood, and avoid the need to use the "any order" version of the assertion in the tests.

It doesn't solve the case of a newer host talking to an older guest, but only in that case can we add in the additional sorting (although it probably needs #1901) then.

Reply:
That is to say, I think the plan is:
- Land this as-is but with the Compose generate fix and changing the tests back
- Add the conditional sorting in a follow-up

Reply:
Since this only causes problems on the new unscoped modifiers, I can fix the Compose code now!

Reply:
Yay! Good idea. https://github.com/cashapp/redwood/commit/f2a3af9d4615917e4a8299ca4f8d70f52194c951